brain/python/circuit.cpp

```diff
@@ -0,0 +1,177 @@
+}
+
+GET_CIRCUIT_PROPERTY_FOR_GIDS(MorphologyTypes)
+GET_CIRCUIT_PROPERTY_FOR_GIDS(ElectrophysiologyTypes)
+
+object Circuit_getMorphologyURIs( const Circuit& circuit, object gids )
+{
+    return toPythonList( circuit.getMorphologyURIs( gidsFromPython( gids )));
+}
+
+#define GET_CIRCUIT_PROPERTY_VALUES(property) \
+    list Circuit_get##property( const Circuit& circuit ) \
+{                                                        \
+    return toPythonList( circuit.get##property( ));      \
+}
+GET_CIRCUIT_PROPERTY_VALUES(MorphologyNames)
+GET_CIRCUIT_PROPERTY_VALUES(ElectrophysiologyNames)
+
+list Circuit_loadMorphologies( const Circuit& circuit,
+                               object gids, Circuit::Coordinates coords )
+{
+    return toPythonList(
+        circuit.loadMorphologies( gidsFromPython( gids ), coords ));
+}
+
+object Circuit_getPositions( const Circuit& circuit, object gids )
+{
+    return toNumpy( circuit.getPositions( gidsFromPython( gids )));
+}
+
+object Circuit_getTransforms( const Circuit& circuit, object gids )
+{
+    object matrices = toNumpy( circuit.getTransforms( gidsFromPython( gids )));
+    // We want the result to be indexed using regular mathematical notation
+    // even if the actual storage is column-major.
+    return matrices.attr( "transpose" )( 0, 2, 1 );
+}
+
+object Circuit_getRotations( const Circuit& circuit, object gids )
+{
+    return toNumpy( circuit.getRotations( gidsFromPython( gids )));
+}
+
+SynapsesWrapper Circuit_getAfferentSynapses(
+    const CircuitPtr& circuit, object gids,
+    const brain::SynapsePrefetch prefetch )
+{
+    return SynapsesWrapper( circuit->getAfferentSynapses( gidsFromPython( gids ),
+                                                          prefetch ),
+                            circuit );
+}
+
+SynapsesWrapper Circuit_getEfferentSynapses(
+    const CircuitPtr& circuit, object gids,
+    const brain::SynapsePrefetch prefetch )
+{
+    return SynapsesWrapper( circuit->getEfferentSynapses( gidsFromPython( gids ),
+                                                          prefetch ),
+                            circuit );
+}
+
+SynapsesWrapper Circuit_getProjectedSynapses(
+    const CircuitPtr& circuit, object pre, object post,
+    const brain::SynapsePrefetch prefetch )
+{
+    return SynapsesWrapper( circuit->getProjectedSynapses(
+                                gidsFromPython( pre ), gidsFromPython( post ),
+                                prefetch ),
+                            circuit );
+}
+
+}
+
+void export_Circuit()
+{
+
+class_< Circuit, boost::noncopyable, CircuitPtr >
+    circuitWrapper( "Circuit", no_init );
+
+scope circuitScope = circuitWrapper;
+
+enum_< Circuit::Coordinates >( "Coordinates" )
+    .value( "global_", Circuit::Coordinates::global )
+    .value( "local", Circuit::Coordinates::local )
+    ;
+
+circuitWrapper
+    .def( "__init__", make_constructor( createCircuitFromURI ))
```

Comment:
One thing I'd would do is to replace presynaptic by pre in brain.Synapse. I didn't do it to respect the C++ name, but once we integrate the automatic documentation Doxygen->docstrings->Sphinx process we can change the names more freely. Just be aware that whenever there's a cross references between methods in Doxygen, the documentation consistency for Python needs to be ensured manually.